### ISSUE TITLE
Caching for `list_accounts` and `list_tax_codes`

### ISSUE THREAD

**Issue by lasuk** [macxred/cashctrl_api#26]
**Description:**
I would like to request a feature that implements caching for the `list_accounts` and `list_tax_codes` methods in the `CashCtrlClient` class. This enhancement aims to reduce the number of API calls by caching the data on the first lookup and reusing the cached data for subsequent calls within a specified timeout period.

**Proposed Solution:**
A new class, `CachedCashCtrlClient`, extends `CashCtrlClient` and overrides the `list_accounts` and `list_tax_codes` methods to include caching functionality. The cache timeout can be configured during class initialization or adjusted later through a property.

In addition, I propose to provide lookup functions to map account numbers to account ids, or tax code names to tax ids, and vice versa.
- `account_to_id(account: float) -> int`
- `account_from_id(id: int) -> float`
- `tax_code_to_id(code: str) -> int`
- `tax_code_from_id(id: int) -> str`

A first implementation is available in the `feat/cache` branch and can be reviewed here:
[CachedCashCtrlClient Implementation](https://github.com/macxred/cashctrl_api/blob/23fb05154/cashctrl_api/cashed_client.py)

**Details:**

- **Initialization:** Accepts an optional `cache_timeout` parameter (default is 300 seconds).
- **Methods Overridden:** `list_accounts`, `list_tax_codes`.
- **Caching Logic:**
  - Data is cached on the first call.
  - Cached data is reused if the cache is not older than the specified timeout.
  - Cache expiration is checked using the `_is_expired` method.
- **Configuration:** The cache timeout can be set during initialization or updated using the `cache_timeout` property.

**Usage Example:**

```python
from cashctrl_api.cashed_client import CachedCashCtrlClient

# Initialize with a cache timeout of 600 seconds (10 minutes)
client = CachedCashCtrlClient(organisation='my_org', api_key='my_api_key', cache_timeout=600)

# Fetch accounts (data will be cached)
accounts = client.list_accounts()

# Fetch tax codes (data will be cached)
tax_codes = client.list_tax_codes()

# Lookup account ID by account number and vice versa
account_id = client.account_to_id(1000)
account = client.account_from_id(1)
```

**Comment by lasuk** [comment]
Let's also add currencies to the cached objects. Currency lookup is also a frequent trigger for an (avoidable) API request.

Steps:
- Add method `CashCtrlClient.list_currencies()`, similar to `CashCtrlClient.list_tax_codes()`.
- Extend `CachedCashCtrlClient` to include `list_currencies()` and the related mapping functions `currency_from_id` and `currency_to_id`.

**Comment by lasuk** [comment]
Implemented as #29
